packages/grid/x-data-grid/src/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -212,6 +212,8 @@ const getFilterCallbackFromItem = (
   };
 };
 
+let filterItemsApplierId = 1;
```

Comment:
If a function is byte-for-byte identical to another, some JS engines re-use the same inline cache data & function info, and in our case it would quickly make our filter functions megamorphic which would be bad for performance. We replace some of the bytes inside our function body with this to prevent that.

Reply:
Interesting, did you anticipate this potential issue or did you actually experience it?

Reply:
Anticipate, I haven't observed it